bindings/rust/src/lazy_reader.rs

```diff
@@ -0,0 +1,488 @@
+    ) -> Result<Cursor, Error> {
+        let mut cur2 = self.clone();
+        let item_count = self.unpack_number()?;
+        if item_index >= item_count {
+            Err(Error::OutOfBound(format!(
+                "item_index({}) >= item_count({})",
+                item_index, item_count
+            )))
+        } else {
+            let offset = calculate_offset(item_size, item_index, NUM_T_SIZE);
+            cur2.add_offset(offset);
+            cur2.size = item_size;
+            cur2.validate()?;
+            Ok(cur2)
+        }
+    }
+
+    pub fn dynvec_slice_by_index(&self, item_index: usize) -> Result<Cursor, Error> {
+        let mut res = self.clone();
+        let mut temp = self.clone();
+        let total_size = self.unpack_number()?;
+        temp.add_offset(NUM_T_SIZE);
+        let item_count = temp.get_item_count()?;
+        if item_index >= item_count {
+            return Err(Error::OutOfBound(format!(
+                "item_index({}) >= item_count({})",
+                item_index, item_count
+            )));
+        }
+        temp.offset = self.offset;
+        let temp_offset = calculate_offset(NUM_T_SIZE, item_index + 1, 0);
+        temp.add_offset(temp_offset);
+        let item_start = temp.unpack_number()?;
+        if (item_index + 1) == item_count {
+            res.offset = self.offset;
+            res.add_offset(item_start);
+            res.size = total_size;
+            res.sub_size(item_start)
+        } else {
+            temp.offset = self.offset;
+            let calc_offset = calculate_offset(NUM_T_SIZE, item_index + 2, 0);
+            temp.add_offset(calc_offset);
+
+            let item_end = temp.unpack_number()?;
+            res.offset = self.offset;
+            res.add_offset(item_start);
+            res.size = item_end;
+            res.sub_size(item_start);
+        }
+        res.validate()?;
+        Ok(res)
+    }
+
+    pub fn table_slice_by_index(&self, field_index: usize) -> Result<Cursor, Error> {
+        self.dynvec_slice_by_index(field_index)
+    }
+
+    pub fn fixvec_slice_raw_bytes(&self) -> Result<Cursor, Error> {
+        let mut res = self.clone();
+        res.add_offset(NUM_T_SIZE);
+        res.size = self.unpack_number()?;
+        res.validate()?;
+        Ok(res)
+    }
+
+    pub fn convert_to_array(&self) -> Result<Cursor, Error> {
+        Ok(self.clone())
+    }
+
+    pub fn convert_to_rawbytes(&self) -> Result<Cursor, Error> {
+        self.fixvec_slice_raw_bytes()
+    }
+
+    pub fn union_unpack(&self) -> Result<Union, Error> {
+        let item_id = self.unpack_number()?;
+        let mut cursor = self.clone();
+        cursor.add_offset(NUM_T_SIZE);
+        cursor.sub_size(NUM_T_SIZE);
+        cursor.validate()?;
+        Ok(Union { item_id, cursor })
+    }
+}
+
+fn calculate_offset(item_size: usize, item_count: usize, offset: usize) -> usize {
+    let res = item_size.checked_mul(item_count).unwrap();
+    res.checked_add(offset).unwrap()
```

Comment:
same issue as `add_offset`, suggest to returning error.

Reply:
see above.